### PR TITLE
fix(MeshAccessLog): strengthen validation for MeshAccessLog and MeshGateway

### PR DESCRIPTION
**Pull request by michaelbeaumont**
This is a breaking change. The access logging is configured at the gateway listener level, which makes it impossible to configure depending on the traffic destination.

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by lahabana** [comment]
The Changelog still needs to be a conventional commit. Isn't it more a change to `UPGRADE.md` that you want?

**Comment by michaelbeaumont** [comment]
> The Changelog still needs to be a conventional commit. Isn't it more a change to UPGRADE.md that you want?

Yeah you're right of course

File: UPGRADE.md [hunk 1]
```diff
@@ -8,6 +8,12 @@ does not have any particular instructions.
 
 ## Upgrade to `2.9.x`
 
+### MeshAccessLog
+
+Policies targeting `spec.targetRef.kind: MeshGateway` can now only target `kind: Mesh` in
+`to[].targetRef`. Previously this was allowed but the resulting configuration
+was ambiguous and nondeterministic.
```
Comment (slonka) [inline UPGRADE.md:15]: I don't understand it, what do you mean by `this`? Shouldn't it be:

```suggestion
Policies targeting `spec.targetRef.kind: MeshGateway` can now only target `kind: Mesh` in
`to[].targetRef`. Previously MeshService, MeshExternalService, MeshMultiZoneService were allowed but the resulting configuration
was ambiguous and nondeterministic.
```

**Review by slonka — COMMENTED** [review]
One thing from changelog seems off to me.

**Review by slonka — APPROVED** [review]

**Pull request merged**